### PR TITLE
feat: add @kampus/email

### LINKED ISSUE #456: feat: add transactional emails using react-email
### Describe the feature you would like to see added

- we used `react-email` with remix before, so nothing new here: https://github.com/kamp-us/monorepo/blob/dev/deprecated/pano/app/features/email/components/Registration.tsx
- we will build upon #455 

### Describe alternatives you have considered

_No response_

### Additional context

_No response_

### LINKED ISSUE #463: feat: use resend.com for email management
### Describe the feature you would like to see added

Use resend.com to send emails. Their free plan might not be good enough (100 mail per day cap might be easy to hit for us since we are planning to utilize magic-links via email for logging people in), but the pro plan is pretty cheap considering it allow up to 50k emails monthly, and there is no daily cap.

Biggest advantage is that they are the creators of `react-email` which we are already using. They have special api to send emails using react components (well, react-email components) and have support for next.js app router out-of-the-box: https://github.com/resendlabs/resend-nextjs-app-router-example/blob/main/app/api/send/route.ts

```typescript
// components/EmailTemplate.tsx
import * as React from "react";

interface EmailTemplateProps {
  firstName: string;
}

export const EmailTemplate: React.FC<Readonly<EmailTemplateProps>> = ({
  firstName,
}) => (
  <div>
    <h1>Welcome, {firstName}!</h1>
  </div>
);

export default EmailTemplate;

// app/email/send/route.ts
import { EmailTemplate } from '../../../components/EmailTemplate';
import { NextResponse } from 'next/server';
import { Resend } from 'resend';

const resend = new Resend(process.env.RESEND_API_KEY);

export async function POST() {
  try {
    const data = await resend.emails.send({
      from: 'onboarding@resend.dev',
      to: 'delivered@resend.dev',
      subject: "Hello world",
      react: EmailTemplate({ firstName: "John" }),
    });

    return NextResponse.json(data);
  } catch (error) {
    return NextResponse.json({ error });
  }
}
```

### Describe alternatives you have considered

#455 

### Additional context

@0bl suggested that using resend.com might be a better idea mainly because mailchannels, which vercel-email (#455) uses under the hood, could be a trouble in the future for spams.

### PR DESCRIPTION
**Pull request by usirin**
Closes #456 
Closes #463 

Make a `POST` request to `/postaci/send` with following json payload:

```json
{
    "email": {
        "to": "kampus-test@resend.dev",
        "subject": "TEST - kamp.us'e hoşgeldin"
    },
    "kampusEmail": {
        "type": "magic-link",
        "props": {
            "url": "https://pasaport.kamp.us"
        }
    }
}
```

### CODE REVIEW

File: apps/kampus/package.json [hunk 1]
```diff
@@ -15,6 +15,7 @@
     "relay-compiler": "15.0.0"
   },
   "dependencies": {
+    "@kampus/email": "0.0.0",
```
Comment (puskuruk) [inline apps/kampus/package.json:18]: Official semver docs does recommend starting at 0.1.0

https://semver.org/#how-should-i-deal-with-revisions-in-the-0yz-initial-development-phase

Comment (Ketcap) [reply apps/kampus/package.json:18]: Actually if it's internal (monorepo) package it should be "@kampus/email: *"

File: packages/email/package.json [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+  "description": "",
+  "main": "index.ts",
+  "types": "index.ts",
+  "scripts": {
+    "dev": "email dev -p 3050"
+  },
+  "author": "",
+  "license": "MIT",
+  "dependencies": {
+    "@kampus/tailwind": "0.0.0",
```
Comment (Ketcap) [inline packages/email/package.json:13]: It's a monorepo dependency it should be marked as "*" 
```suggestion
    "@kampus/tailwind": "*",
```

File: apps/kampus/package.json (hunk 1 above)
Comment (puskuruk) [reply apps/kampus/package.json:18]: Interesting. I always try to version my packages, no matter if they're internal or not. Why do you suggest using `*` as a version number?

Comment (Ketcap) [reply apps/kampus/package.json:18]: [Check here](https://turbo.build/repo/docs/handbook/sharing-code/internal-packages#what-makes-a-package-internal) , internal packages are not versioned or published. If you change internal package to version "0.2.0" but one of your app uses "0.1.0" which is not something published or can be accessed.

Comment (Ketcap) [reply apps/kampus/package.json:18]: Their suggestion to use "*" as well https://turbo.build/repo/docs/handbook/sharing-code/internal-packages#3-import-the-package

Comment (puskuruk) [reply apps/kampus/package.json:18]: I didn't know that. Thank you 🙏

Comment (usirin) [reply apps/kampus/package.json:18]: > Actually if it's internal (monorepo) package it should be "@kampus/email: *"

wait, it is `"@kampus/email": "*" right?

Comment (usirin) [reply apps/kampus/package.json:18]: BTW, ty so much for the explanation @Ketcap i will open a PR to fix this in each package.

File: packages/email/emails/MagicLink.tsx [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+export const MagicLink = ({ url = "https://pasaport.kamp.us/magic-link" }: MagicLinkProps) => {
+  return (
+    <Email
+      heading={<EmailHeading>Selam!</EmailHeading>}
+      preview={<EmailPreview>kamp.us&apos;e hoşgeldin</EmailPreview>}
+    >
+      <div className="flex flex-col gap-4">
+        {url && (
+          <Link href={url} target="_blank" className="font-sans text-sm text-purple-700 underline">
+            kamp.us&apos;e giriş yapmak icin tıkla
```
Comment (cansirin) [inline packages/email/emails/MagicLink.tsx:19]:
```suggestion
           için tıkla
```

Comment (cansirin) [inline packages/email/emails/MagicLink.tsx:19]:
```suggestion
            kamp.us&apos;e giriş yapmak için tıkla
```

**Review by cansirin — APPROVED** [review]

**Comment by usirin** [comment]
TY so much for the review @Ketcap @puskuruk 

Merging this to move forward with development. Please review and open report bugs if you see.

**Pull request merged**